Stripe/StripeiOSTests/STPAPIClientStubbedTest.swift

```diff
@@ -38,6 +38,126 @@ class STPAPIClientStubbedTest: APIStubbedTestCase {
+                        return item.name == "\(base)[client_attribution_metadata][merchant_integration_subtype]" && item.value == "mobile"
+                    }
+                    return item.name == "client_attribution_metadata[merchant_integration_subtype]" && item.value == "mobile"
+                }), shouldContainClientAttributionMetadata)
+                XCTAssertEqual(queryItems.contains(where: { item in
+                    if let base {
+                        return item.name == "\(base)[client_attribution_metadata][merchant_integration_version]" && item.value == "stripe-ios/\(StripeAPIConfiguration.STPSDKVersion)"
+                    }
+                    return item.name == "client_attribution_metadata[merchant_integration_version]" && item.value == "stripe-ios/\(StripeAPIConfiguration.STPSDKVersion)"
+                }), shouldContainClientAttributionMetadata)
+                return true
+            } response: { _ in
+                return .init()
+            }
+        }
+
+    func testCreatePaymentMethodWithClientAttributionMetadata() {
+        let sut = stubbedAPIClient()
+        AnalyticsHelper.shared.generateSessionID()
+        stubClientAttributionMetadata()
+        let e = expectation(description: "")
+        sut.createPaymentMethod(with: ._testValidCardValue(), additionalPaymentUserAgentValues: []) { _, _ in
+            e.fulfill()
+        }
+        waitForExpectations(timeout: 10)
+    }
+
+    func testCreateApplePayPaymentMethodWithClientAttributionMetadata() {
+        let sut = stubbedAPIClient()
+        AnalyticsHelper.shared.generateSessionID()
+        stubClientAttributionMetadata()
+        let e = expectation(description: "")
+        StripeAPI.PaymentMethod.create(apiClient: sut, params: StripeAPI.PaymentMethodParams(type: .card)) { _ in
+            e.fulfill()
+        }
+        waitForExpectations(timeout: 10)
+    }
+
+    func testConfirmPaymentIntentWithClientAttributionMetadata() {
+        let sut = stubbedAPIClient()
+        AnalyticsHelper.shared.generateSessionID()
+        stubClientAttributionMetadata(base: "payment_method_data")
+        let e = expectation(description: "")
+        let paymentMethodParams = STPPaymentMethodParams()
+        let paymentIntentParams = STPPaymentIntentParams(clientSecret: "pi_123456_secret_654321")
+        paymentIntentParams.paymentMethodParams = paymentMethodParams
+        sut.confirmPaymentIntent(with: paymentIntentParams) { _, _ in
+            e.fulfill()
+        }
+        waitForExpectations(timeout: 10)
+    }
+
+    func testConfirmPaymentIntentWithoutClientAttributionMetadata() {
+        let sut = stubbedAPIClient()
+        AnalyticsHelper.shared.generateSessionID()
+        stubClientAttributionMetadata(base: "payment_method_data", shouldContainClientAttributionMetadata: false)
+        let e = expectation(description: "")
+        let paymentIntentParams = STPPaymentIntentParams(clientSecret: "pi_123456_secret_654321")
```

Comment:
I wondered why this PI shouldn't have client metadata, maybe we can add a comment explaining (seems like PIs confirmed with a PM id instead of PM params should not send client metadata? B/c the PM is intended to carry the metadata?).